executable-spec/src/Cardano/Ledger/Spec/State/StakeDistribution.hs

```diff
@@ -21,45 +36,118 @@ import qualified Ledger.Core as Core
 import           Cardano.Ledger.Spec.Classes.Hashable (Hash, Hashable)
 import           Cardano.Ledger.Spec.Classes.HasSigningScheme (HasSigningScheme,
                      VKey)
-import           Cardano.Ledger.Spec.Classes.Indexed (Indexed)
+import           Cardano.Ledger.Spec.Classes.Indexed (Indexed, Key, Value,
+                     lookup)
+import           Cardano.Ledger.Spec.STS.Common.Compact (CompactHash)
 
 
-newtype StakeDistribution p = StakeDistribution (Map (Hash p (VKey p)) Data.Stake)
+data StakeDistributionCompact =
+  StakeDistributionCompact
+  { stakeMapCmp    :: !(Map CompactHash Data.Stake)
```

Comment:
I think we don't need to change this. The `Hash` instance can simply use a compact bytestring without this module needing to know anything about it.